File: 00003/solution.ts
function biggestPrime(n: number):number | null {
    let largest:number = -1;
    if (n <= 1) {
        return null;
    }

    while (n%2 === 0) {
        largest = 2;
        n = Math.floor(n / 2);
    } 

    for (let p = 3; p * p <= n; p+= 2) {
        while (n % p === 0) {
            largest = p;
            n = Math.floor(n / p);
        }
    }

    if (n > 1) largest = n;

    return largest
}

console.log(biggestPrime(600851475143))

File: 00003/solution.py
import math

def find_biggest_prime_factor(n: int) -> int:
    if n<= 1:
        return None
    
    big_prime = -1

    while n % 2 == 0:
        big_prime = 2
        n = n //  2
    
    p = 3
    while p * p <= n:
        while n % p == 0:
            big_prime = p
            n = n // p
        p += 2
    
    if n > 1:
        big_prime = n

    return big_prime

print(find_biggest_prime_factor(600851475143))